### ISSUE TITLE
22.3.5 + master: pytest is failing

### ISSUE THREAD

**Issue by kloczek** [flyingcircusio/pycountry#169]
I'm packaging your module as an rpm package so I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix> using 'installer` module
- run pytest with $PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>
- build is performed in env which is *`cut off from access to the public network`* (pytest is executed with `-m "not network"`)

Here is pytest output:
<details>

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-pycountry-22.3.5-9.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-pycountry-22.3.5-9.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -m 'not network'
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.8.18, pytest-7.4.2, pluggy-1.3.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pycountry-22.3.5
configfile: pytest.ini
collected 21 items

src/pycountry/tests/test_general.py ........FF.F........F                                                                                                                             [100%]

========================================================================================= FAILURES ==========================================================================================
______________________________________________________________________________________ test_currencies ______________________________________________________________________________________
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 262, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/usr/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
  File "/usr/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 177, in pytest_runtest_call
    raise e
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 169, in pytest_runtest_call
    item.runtest()
  File "/usr/lib/python3.8/site-packages/_pytest/python.py", line 1797, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/usr/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 113, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/usr/lib/python3.8/site-packages/_pytest/python.py", line 199, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/home/tkloczko/rpmbuild/BUILD/pycountry-22.3.5/src/pycountry/tests/test_general.py", line 117, in test_currencies
    assert len(pycountry.currencies) == 170
AssertionError: assert 181 == 170
 +  where 181 = len(<pycountry.Currencies object at 0x7f9390ad2fd0>)
 +    where <pycountry.Currencies object at 0x7f9390ad2fd0> = pycountry.currencies
______________________________________________________________________________________ test_languages _______________________________________________________________________________________
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 262, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/usr/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
  File "/usr/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 177, in pytest_runtest_call
    raise e
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 169, in pytest_runtest_call
    item.runtest()
  File "/usr/lib/python3.8/site-packages/_pytest/python.py", line 1797, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/usr/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 113, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/usr/lib/python3.8/site-packages/_pytest/python.py", line 199, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/home/tkloczko/rpmbuild/BUILD/pycountry-22.3.5/src/pycountry/tests/test_general.py", line 127, in test_languages
    assert len(pycountry.languages) == 7847
AssertionError: assert 7910 == 7847
 +  where 7910 = len(<pycountry.Languages object at 0x7f9390ad20d0>)
 +    where <pycountry.Languages object at 0x7f9390ad20d0> = pycountry.languages
_______________________________________________________________________________________ test_locales ________________________________________________________________________________________
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 262, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/usr/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
  File "/usr/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 177, in pytest_runtest_call
    raise e
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 169, in pytest_runtest_call
    item.runtest()
  File "/usr/lib/python3.8/site-packages/_pytest/python.py", line 1797, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/usr/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 113, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/usr/lib/python3.8/site-packages/_pytest/python.py", line 199, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/home/tkloczko/rpmbuild/BUILD/pycountry-22.3.5/src/pycountry/tests/test_general.py", line 154, in test_locales
    german = gettext.translation(
  File "/usr/lib64/python3.8/gettext.py", line 588, in translation
    raise FileNotFoundError(ENOENT,
FileNotFoundError: [Errno 2] No translation file found for domain: 'iso_3166'
________________________________________________________________________________ test_has_version_attribute _________________________________________________________________________________
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 262, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/usr/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
  File "/usr/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 177, in pytest_runtest_call
    raise e
  File "/usr/lib/python3.8/site-packages/_pytest/runner.py", line 169, in pytest_runtest_call
    item.runtest()
  File "/usr/lib/python3.8/site-packages/_pytest/python.py", line 1797, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/usr/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 113, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/usr/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/usr/lib/python3.8/site-packages/_pytest/python.py", line 199, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/home/tkloczko/rpmbuild/BUILD/pycountry-22.3.5/src/pycountry/tests/test_general.py", line 246, in test_has_version_attribute
    assert pycountry.__version__ != "n/a"
AssertionError: assert 'n/a' != 'n/a'
 +  where 'n/a' = pycountry.__version__
------------------------------------------------------- generated xml file: /home/tkloczko/rpmbuild/BUILD/pycountry-22.3.5/report.xml -------------------------------------------------------
================================================================================== short test summary info ==================================================================================
FAILED src/pycountry/tests/test_general.py::test_currencies - assert 181 == 170
FAILED src/pycountry/tests/test_general.py::test_languages - assert 7910 == 7847
FAILED src/pycountry/tests/test_general.py::test_locales - FileNotFoundError: [Errno 2] No translation file found for domain: 'iso_3166'
FAILED src/pycountry/tests/test_general.py::test_has_version_attribute - AssertionError: assert 'n/a' != 'n/a'
=============================================================================== 4 failed, 17 passed in 0.55s ================================================================================
```
</details>

Here is list of installed modules in build env
<details>

```console
Package            Version
------------------ -------
build              1.0.0
cppclean           0.13
distro             1.8.0
exceptiongroup     1.1.3
gpg                1.21.0
importlib-metadata 6.8.0
iniconfig          2.0.0
installer          0.7.0
packaging          23.2
pkg                0.0.0
pluggy             1.3.0
poetry-core        1.7.0
pyproject_hooks    1.0.0
pytest             7.4.2
python-dateutil    2.8.2
six                1.16.0
tomli              2.0.1
wheel              0.41.2
zipp               3.17.0
```
</details>

**Comment by zware** [comment]
This will be addressed by #140/#164.

**Comment by zware** [comment]
There may be an additional fix needed for the version test, I can look into that once the other failures have been fixed.

**Comment by kloczek** [comment]
👍 
FYI: I've added temporary in my build procedure those units to --deselect list (as looks like it is test suite issue .. no rush)

**Comment by kloczek** [comment]
Looking on list of installed files Looks like test suite files are installed.
Easiest way to fix this will be move pycountry/tests to tests/.

**Comment by yezz123** [comment]
I made a hard fork for pycountry, maybe you can use it https://github.com/yezz123/isocountry/ (I will support it next in https://github.com/pydantic/pydantic-extra-types/pull/101 )

**Comment by zware** [comment]
Fixed in #140/#164.

**Comment by kloczek** [comment]
Just tested 23.12.7 and looks like pytest is OK now (thank you) however I found that content of the src/pycountry/tests is packaged into .whl archive.
Probably simplest solution would be move src/pycountry/tests to tests/.
